daemon/src/model/cfd.rs

```diff
@@ -1234,20 +1234,30 @@ pub fn calculate_long_liquidation_price(leverage: Leverage, price: Price) -> Pri
     price * leverage / (leverage + 1)
 }
 
+pub fn calculate_profit(payout: SignedAmount, margin: SignedAmount) -> (SignedAmount, Percent) {
+    let profit = payout - margin;
+
+    let profit_sats = Decimal::from(profit.as_sat());
+    let margin_sats = Decimal::from(margin.as_sat());
+    let percent = dec!(100) * profit_sats / margin_sats;
```

Comment:
@thomaseizinger is it OK to use the primitive impls over `checked_mul` and `checked_div` here?
Otherwise we would need to return a `Result` again.

Reply:
They will panic on overflow / underflow in debug mode and overflow / underflow in release builds. If you are okay with that, then you can use them.

Reply:
I think these calculations with `SignedAmount` should always fit into the `Decimal` type. I don't think we can effectively run into an under-/overflow here. I'll keep as is.